cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionedFileSet.java

```diff
@@ -109,15 +115,34 @@
    * in the iterator returned in the PartitionConsumerResult.
    *
    * @param partitionConsumerState the state from which to start consuming from
-   * @param partitionFilter a filter which must match the partitions to be consumed
    * @param limit number of partitions, which once reached, will not add add more partitions committed by other
    *              transactions; the limit is checked after adding consuming all partitions of a transaction, so
    *              the total number of consumed partitions may be greater than this limit
+   * @param predicate a predicate which determines the partitions to be consumed
    * @return {@link PartitionConsumerResult} which holds the state of consumption as well as an iterator to the consumed
    *         {@link Partition}s
+   * @Deprecated Use {@link #getPartitionConsumer(StatePersistor)} instead.
    */
+  @Deprecated
   PartitionConsumerResult consumePartitions(PartitionConsumerState partitionConsumerState,
-                                            PartitionFilter partitionFilter, int limit);
+                                            int limit, Predicate<PartitionDetail> predicate);
+
+  /**
+   * Provides access to a {@link PartitionConsumer} for consuming {@link Partition}s of this dataset.
+   *
+   * @param statePersistor defines how the state of the PartitionConsumer will be maintained
+   * @return an instance of {@link PartitionConsumer} for incremental processing
+   */
+  PartitionConsumer getPartitionConsumer(StatePersistor statePersistor);
+
+  /**
+   * Provides access to a {@link PartitionConsumer} for consuming {@link Partition}s of this dataset.
+   *
```

Comment:
This should explain what the configuration parameters passed to the PartitionConsumer are for; why use this instead of the previous method?